src/components/LinkedAccounts/OpeningBalanceModal/useUpdateOpeningBalanceAndDate.ts

```diff
@@ -50,84 +108,47 @@ export function useUpdateOpeningBalanceAndDate(
     return errors
   }
 
-  const updateData = ({
-    account: { id: accountId },
-    openingBalance: openingBalance,
-    openingDate: openingDate,
-  }: AccountFormBoxData) => {
-    if (!openingDate || !openingBalance) {
-      return
-    }
-
-    return Layer.updateOpeningBalance(
-      auth?.apiUrl ?? '',
-      auth?.access_token,
-      {
-        params: {
-          businessId,
-          accountId,
-        },
-        body: {
-          effective_at: openingDate.toISOString(),
-          balance: openingBalance,
-        },
-      },
+  return useSWRMutation(
+    () => buildKey({ access_token: auth?.access_token, apiUrl: auth?.apiUrl, businessId, data }),
+    ({ accessToken, apiUrl, businessId, data }) => (
+      Promise.allSettled(
+        data.map(
+          ({ accountId, openingDate, openingBalance }) => {
+            const errors = validate({ accountId, openingDate, openingBalance })
+            if (errors.length > 0) {
+              return Promise.reject(
+                new Error('Invalid data', {
+                  cause: {
+                    type: 'validation',
+                    errors,
+                    accountId,
+                  },
+                }),
```

Comment:
I'm extending the error content so it's easier to distinct validation and failed API calls